custom_components/moonraker/sensor.py

```diff
@@ -452,3 +498,22 @@ def convert_time(time_s):
     return (
         f"{round(time_s // 3600)}h {round(time_s % 3600 // 60)}m {round(time_s % 60)}s"
     )
+
+
+def calculate_current_layer(data):
```

Comment:
Nit pick. I would move the function in the class Like `empty_result_when_not_printing`

Reply:
I don't know, it seems that it fit with the other "calculate" method like eta and pct_job